code/controllers/DMSDocumentCartCheckoutPage_Controller.php

```
@@ -0,0 +1,223 @@
+        $fields->replaceField('DeliveryAddressCountry', CountryDropdownField::create(
+            'DeliveryAddressCountry',
+            _t('DMSDocumentCartCheckoutPage.RECEIVER_COUNTRY', 'Country')
+        )->setValue('NZ'));
+
+        $requiredFields = array(
+            'ReceiverName',
+            'ReceiverPhone',
+            'ReceiverEmail',
+            'DeliveryAddressLine1',
+            'DeliveryAddressCountry',
+            'DeliveryAddressPostCode',
+        );
+        foreach ($fields as $field) {
+            if (in_array($field->name, $requiredFields)) {
+                $field->addExtraClass('requiredField');
+            }
+        }
+        $validator = RequiredFields::create($requiredFields);
+        $actions = FieldList::create(
+            FormAction::create(
+                'doRequestSend',
+                _t('DMSDocumentCartCheckoutPage.SEND_ACTION', 'Send your request')
+            )
+        );
+
+        $form = Form::create(
+            $this,
+            'DMSDocumentRequestForm',
+            $fields,
+            $actions,
+            $validator
+        );
+
+        if ($receiverInfo = $this->getCart()->getReceiverInfo()) {
+            $form->loadDataFrom($receiverInfo);
+        }
+
+        $form->setTemplate('DMSDocumentRequestForm');
+        $this->extend('updateDMSDocumentRequestForm', $form);
+
+        return $form;
+    }
+
+    /**
+     * Sends an email to both the configured recipient as well as the requester. The
+     * configured recipient is bcc'ed to the email in order to fulfill it.
+     *
+     * To extend use the following from within an Extension subclass:
+     *
+     * <code>
+     * public function updateSend($email)
+     * {
+     *     // Do something here
+     * }
+     * </code>
+     * @return mixed
+     *
+     * @throws DMSDocumentCartException
+     */
+    public function send()
+    {
+        $member = $this->CartEmailRecipient();
+        if ($member->exists()) {
```

If you reverse this you can throw the exception early and reduce indentation for the rest of the method